sdk/aqueduct/decorator.py

```diff
@@ -71,17 +73,19 @@ def wrap_spec(
         description:
             The description for this operator.
     """
-    if len(input_artifacts) < 1:
-        raise Exception("Transformation requires at least one input artifact.")
-
     for artifact in input_artifacts:
         if artifact._from_flow_run:
             raise InvalidUserActionException(
                 "Artifact fetched from flow run can not be reused for computation"
             )
 
-    dag = input_artifacts[0]._dag
-    api_client = input_artifacts[0]._api_client
+    if len(input_artifacts) == 0:
+        print("This operator takes no artifact input.")
+        dag = dag_module.__GLOBAL_DAG__
+        api_client = api_client_module.__GLOBAL_API_CLIENT__
+    else:
+        dag = input_artifacts[0]._dag
+        api_client = input_artifacts[0]._api_client
```

Comment:
I'm not sure we should even keep the else statement once we have global variables.